packages/nextjs/utils/satisfies.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * Simulates behavior of TypeScript 4.9's "satisfies" operator
+ * @example
+ *   const getServerSideProps = satisfies<GetServerSideProps>()(async (context) => ({ props: { myProp1: true } });
+ *   type SSRProps = AsyncReturnType<getServerSideProps>['props']; // { myProp1: true }
+ */
+export const satisfies =
```

Comment:
I'll create an issue to upgrade version of TS, I don't know if there should be anything that is breaking but would be good to check things before yolo upgrade

Reply:
sorry didn't realize 4.9 is not public yet, I made an [issue](https://github.com/FormidableLabs/nextjs-sanity-fe/issues/154) regardless to upgrade when it's generally available

Reply:
Yes, good idea. I was going to upgrade to the beta, but I only needed `satisfies` for a few places, so I went with this workaround version. It will be nice to remove all that extra `<>()()` !